File: rtsp/rtsp.cc
#include <iostream>
#include <opencv4/opencv2/opencv.hpp>

using namespace cv;

int main(){
	
	auto receive = VideoCapture("test.mp4");

	auto send = VideoWriter("appsrc ! videoconvert ! x264enc tune=zerolatency bitrate=500 speed-preset=superfast ! rtph264pay ! udpsink host=0.0.0.0 port    =5000", CAP_GSTREAMER, 0, 20, Size(320, 240), true);
	

	if(!receive.isOpened())  // check if we succeeded
		return -1;

	Mat frame;

	while (true) {

		receive >> frame; // get a new frame from camera

		resize(frame, frame, Size(320, 240));

		send.write(frame);

	}

}

File: rtsp/rtsp.py
import cv2


cap_send = cv2.VideoCapture('test.mp4')

out_send = cv2.VideoWriter('appsrc ! videoconvert ! x264enc tune=zerolatency bitrate=500 speed-preset=superfast ! rtph264pay ! udpsink host=127.0.0.1 port=5000',cv2.CAP_GSTREAMER, 0, 20, (320,240), True)

if not cap_send.isOpened() or not out_send.isOpened():
    print('VideoCapture or VideoWriter not opened')
    exit(0)

while True:
    
    ret,frame = cap_send.read()

    if not ret:
    
        print('empty frame')
        
        break

    frame = cv2.resize(frame, (320, 240))

    out_send.write(frame)

    if cv2.waitKey(1)&0xFF == ord('q'):
        break

cap_send.release()
out_send.release()
